src/humanloop/otel/processor.py

```diff
@@ -0,0 +1,194 @@
+                self._children[span.parent.span_id].append(span)
+        # Pass the Span to the Exporter
+        self.span_exporter.export([span])
+
+
+def _is_instrumentor_span(span: ReadableSpan) -> bool:
+    """Determine if the Span contains information of interest for Spans created by Humanloop decorators."""
+    # At the moment we only enrich Spans created by the Prompt decorators
+    # As we add Instrumentors for other libraries, this function must
+    # be expanded
+    return is_llm_provider_call(span=span)
+
+
+def _process_span_dispatch(span: ReadableSpan, children_spans: list[ReadableSpan]):
+    hl_file = read_from_opentelemetry_span(span, key=HL_FILE_OT_KEY)
+
+    if "prompt" in hl_file:
+        _process_prompt(prompt_span=span, children_spans=children_spans)
+        return
+    elif "tool" in hl_file:
+        _process_tool(tool_span=span, children_spans=children_spans)
+        return
+    elif "flow" in hl_file:
+        _process_flow(flow_span=span, children_spans=children_spans)
+        return
+    else:
+        logging.error("Invalid span type")
+
+
+def _process_prompt(prompt_span: ReadableSpan, children_spans: list[ReadableSpan]):
+    if len(children_spans) == 0:
+        return
+    child_span = children_spans[0]
+    assert is_llm_provider_call(child_span)
+    _enrich_prompt_span_file(prompt_span, child_span)
+    _enrich_prompt_span_log(prompt_span, child_span)
+
+
+def _process_tool(tool_span: ReadableSpan, children_spans: list[ReadableSpan]):
+    tool_log = read_from_opentelemetry_span(tool_span, key=HL_LOG_OT_KEY)
+    if tool_span.start_time:
+        tool_log["start_time"] = tool_span.start_time / 1e9
+    if tool_span.end_time:
+        tool_log["end_time"] = tool_span.end_time / 1e9
+        tool_log["created_at"] = tool_span.end_time / 1e9
+
+    write_to_opentelemetry_span(
+        span=tool_span,
+        key=HL_LOG_OT_KEY,
+        value=tool_log,
+    )
+
+
+def _process_flow(flow_span: ReadableSpan, children_spans: list[ReadableSpan]):
+    # TODO: Use children_spans in the future
+    flow_log = read_from_opentelemetry_span(flow_span, key=HL_LOG_OT_KEY)
+    if flow_span.start_time:
+        flow_log["start_time"] = flow_span.start_time / 1e9
+    if flow_span.end_time:
+        flow_log["end_time"] = flow_span.end_time / 1e9
+        flow_log["created_at"] = flow_span.end_time / 1e9
+
+    write_to_opentelemetry_span(
+        span=flow_span,
+        key=HL_LOG_OT_KEY,
+        value=flow_log,
+    )
+
+
+def _enrich_prompt_span_file(prompt_span: ReadableSpan, llm_provider_call_span: ReadableSpan):
+    hl_file: dict[str, Any] = read_from_opentelemetry_span(prompt_span, key=HL_FILE_OT_KEY)
+    gen_ai_object: dict[str, Any] = read_from_opentelemetry_span(llm_provider_call_span, key="gen_ai")
+    llm_object: dict[str, Any] = read_from_opentelemetry_span(llm_provider_call_span, key="llm")
+
+    prompt = hl_file.get("prompt")
+    if not prompt:
```

Comment:
This feels like a maintenance hazard, is there anyway to somehow reuse the types from the SDK so that we have a more exhaustive list? e.g. there are some things missing here that i would expect to be required:

tools, attributes

Reply:
1. I've replied to the tools point above with a question.
2. I will add the attributes
3. I can't use PromptKernelRequest since it's a Pydantic class configured as frozen and can't be modified. I've added cast to PromptKernelRequestParams which is a `TypedDict` and a `PromptKernelRequest.model_validate`  to check if the enriched kernel is as expected